wasm-chain-integration/src/v1/tests.rs

```diff
@@ -0,0 +1,839 @@
+        );
+    } else {
+        bail!("Write should've returned Ok(0)");
+    }
+    ensure!(
+        state.entry_read(entry, &mut overflowing_buffer, 0) as usize
+            == overflowing_buffer.len() - 1,
+        "Only 2^31 bytes should be read."
+    );
+    let mut energy_supplied = crate::InterpreterEnergy {
+        energy: u64::MAX,
+    };
+    let resize_status =
+        if let Ok(resize_status) = state.entry_resize(&mut energy_supplied, entry, 0) {
+            resize_status
+        } else {
+            bail!("Resizing to empty should not have returned an Err.");
+        };
+    ensure!(resize_status == 1, "Resizing to 0 should have been completed successfully.");
+
+    energy_supplied.energy = 0;
+    ensure!(
+        state
+            .entry_resize(&mut energy_supplied, entry, (crate::constants::MAX_ENTRY_SIZE) as u32)
+            .is_err(),
+        "Resizing without sufficient energy should return an Err."
+    );
+
+    energy_supplied.energy = u64::MAX;
+    let resize_status = if let Ok(resize_status) =
+        state.entry_resize(&mut energy_supplied, entry, crate::constants::MAX_ENTRY_SIZE as u32)
+    {
+        resize_status
+    } else {
+        bail!("Resizing max should not have returned an Err.");
+    };
+    ensure!(
+        resize_status == 1,
+        "Resizing to MAX_ENTRY_SIZE should have been completed successfully."
+    );
+
+    let resize_status = if let Ok(resize_status) = state.entry_resize(
+        &mut energy_supplied,
+        entry,
+        (crate::constants::MAX_ENTRY_SIZE + 1) as u32,
+    ) {
+        resize_status
+    } else {
+        bail!("Resizing MAX_ENTRY_SIZE + 1 should not have returned an Err.");
+    };
+    ensure!(
+        resize_status == 0,
+        "Resizing to MAX_ENTRY_SIZE + 1 should have been completed successfully."
+    );
+
+    ensure!(state.delete_entry(entry) == 1, "Deletion of entry {:?} should return 1", k);
+    let resize_status =
+        if let Ok(resize_status) = state.entry_resize(&mut energy_supplied, entry, 0) {
+            resize_status
+        } else {
+            bail!("Resizing of invalidated entry should not have returned an Err.");
+        };
+    ensure!(resize_status == u32::MAX, "Resizing invalidated entry should return u32::MAX.");
+    Ok(())
+}
+
+#[test]
+/// Test that:
+/// 1. Getting the size of an invalid entry returns u32::MAX.
+/// 2. Deleting an invalidated entry returns u32::MAX.
+/// 3. Looking up an invalid entry returns InstanceStateEntryOption::NEW_NONE.
+/// 4. Delete prefix on non existent key in tree returns 1.
+fn test_size_of_invalid_entry() -> anyhow::Result<()> {
+    let loader = trie::Loader {
+        inner: Vec::<u8>::new(),
+    };
+    let mut m_state = MutableState::initial_state();
+    let mut state = InstanceState::new(0, loader, m_state.get_inner());
+    let entry = if let Ok(entry) = state.create_entry(&[0]) {
+        entry
+    } else {
+        bail!("Entry should be created.");
+    };
```

Comment:
```suggestion
    let entry =  state.create_entry(&[0]).context("Entry should be created.")?;
```

Reply:
Generally I think this style will lead to more readability elsewhere in the tests you wrote by cutting down on the visual noise.